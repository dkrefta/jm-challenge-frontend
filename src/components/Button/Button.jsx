import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Icon from '../Icon/Icon';

const ButtonComponent = props => (
  <Button {...props}>
    {props.children || ' '}
    {props.icon && (
        <Icon tag="arrow-right" />
    )}
    {props.loading && (
        <Icon tag="spinner" />
    )}
  </Button>
);

const Button = styled.button.attrs({
  className: `form-button`
})`
  margin-bottom: 30px; 
  justify-content: space-between; 
  display: flex;
  background: ${({ background }) => background || '#fff'};
  color: ${({ color }) => color || '#9779f1'};
  cursor: ${({ disabled }) => (!!disabled ? 'default' : 'pointer')};
  font-size: ${({ size }) => {
    switch (size) {
      case 'small':
        return '.8rem';
      case 'big':
        return '1.6rem';
      default:
        return '1rem';
    }
  }};
  align-items: center;
  font-weight: 600;
  opacity: ${({ disabled }) => (!!disabled ? 0.5 : 1)};
  outline: none;
  padding: ${({ size }) => {
        return '20px 30px';
    }
  }};
  &:hover:enabled {
    background: ${({ hoverBg }) => hoverBg || '#D3D3D3'};
    color: ${({ hoverColor }) => hoverColor || '#9779f1'};
  }
`;

const IconWrapper = styled.span`
height: 1rem;
width: 1rem;
margin-left: 10px;
`;

export default ButtonComponent;