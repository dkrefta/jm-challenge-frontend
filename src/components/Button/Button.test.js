import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button.jsx';

describe('Button', () => {
 it('should be defined', () => {
   expect(Button).toBeDefined();
 });
 it('should render correctly', () => {
   const tree = shallow(
     <Button name='button test' />
   );
   expect(tree).toMatchSnapshot();
 });
});