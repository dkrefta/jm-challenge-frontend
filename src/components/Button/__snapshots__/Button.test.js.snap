// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonComponent
    name="button test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        " ",
        undefined,
        undefined,
      ],
      "name": "button test",
    },
    "ref": null,
    "rendered": Array [
      " ",
      undefined,
      undefined,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Object {
        "className": "form-button",
      },
      "componentStyle": ComponentStyle {
        "componentId": "Button-lke34x-0",
        "isStatic": false,
        "rules": Array [
          "margin-bottom:30px;justify-content:space-between;display:flex;width:80%;background:",
          [Function],
          ";color:",
          [Function],
          ";cursor:",
          [Function],
          ";font-size:",
          [Function],
          ";align-items:center;font-weight:600;opacity:",
          [Function],
          ";outline:none;padding:",
          [Function],
          "};&:hover:enabled{background:",
          [Function],
          ";color:",
          [Function],
          ";}",
        ],
      },
      "displayName": "Button",
      "render": [Function],
      "styledComponentId": "Button-lke34x-0",
      "target": "button",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          " ",
          undefined,
          undefined,
        ],
        "name": "button test",
      },
      "ref": null,
      "rendered": Array [
        " ",
        undefined,
        undefined,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Object {
          "className": "form-button",
        },
        "componentStyle": ComponentStyle {
          "componentId": "Button-lke34x-0",
          "isStatic": false,
          "rules": Array [
            "margin-bottom:30px;justify-content:space-between;display:flex;width:80%;background:",
            [Function],
            ";color:",
            [Function],
            ";cursor:",
            [Function],
            ";font-size:",
            [Function],
            ";align-items:center;font-weight:600;opacity:",
            [Function],
            ";outline:none;padding:",
            [Function],
            "};&:hover:enabled{background:",
            [Function],
            ";color:",
            [Function],
            ";}",
          ],
        },
        "displayName": "Button",
        "render": [Function],
        "styledComponentId": "Button-lke34x-0",
        "target": "button",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
